mssql_python/pybind/logger_bridge.cpp

```
@@ -0,0 +1,210 @@
+    }
+    
+    // Message was truncated - allocate larger buffer
+    // (This should be rare for typical log messages)
+    std::vector<char> large_buffer(result + 1);
+    va_copy(args_copy, args);
+    // std::vsnprintf is safe here too - proper bounds checking with buffer size
+    std::vsnprintf(large_buffer.data(), large_buffer.size(), format, args_copy);
+    va_end(args_copy);
+    
+    return std::string(large_buffer.data());
+}
+
+const char* LoggerBridge::extractFilename(const char* path) {
+    // Extract just the filename from full path using safer C++ string search
+    if (!path) {
+        return "";
+    }
+    
+    // Find last occurrence of Unix path separator
+    const char* filename = std::strrchr(path, '/');
+    if (filename) {
+        return filename + 1;
+    }
+    
+    // Try Windows path separator
+    filename = std::strrchr(path, '\\');
+    if (filename) {
+        return filename + 1;
+    }
+    
+    // No path separator found, return the whole string
+    return path;
+}
+
+void LoggerBridge::log(int level, const char* file, int line, 
+                      const char* format, ...) {
+    // Fast level check (should already be done by macro, but double-check)
+    if (!isLoggable(level)) {
+        return;
+    }
+    
+    // Check if initialized
+    if (!initialized_ || !cached_logger_) {
+        return;
+    }
+    
+    // Format the message
+    va_list args;
+    va_start(args, format);
+    std::string message = formatMessage(format, args);
+    va_end(args);
+    
+    // Extract filename from path
+    const char* filename = extractFilename(file);
+    
+    // Format the complete log message with [DDBC] prefix for CSV parsing
+    // File and line number are handled by the Python formatter (in Location column)
+    // std::snprintf is safe: always null-terminates, never overflows buffer
+    // DevSkim warning is false positive - this is the recommended safe alternative
+    char complete_message[4096];
```

If `message.c_str()` is longer than 4090 bytes, it's silently truncated.. I don't see any warning to the user that log data was lost. In troubleshooting scenarios, critical diagnostic info may be missing. Can we atleast warn the customer?